client/template.go

```diff
@@ -131,18 +131,31 @@ func (payload *TemplateCreatePayload) Invalidate() error {
 		return errors.New("can't define terragrunt version for non-terragrunt template")
 	}
 
-	if payload.Type == TERRAGRUNT && payload.TerragruntVersion == "" {
-		return errors.New("must supply terragrunt version")
+	if payload.Type == TERRAGRUNT {
+		if payload.TerragruntVersion == "" {
+			return errors.New("must supply terragrunt version")
+		}
+
+		// The provider implicitly defaults to "opentofu".
+		if payload.TerragruntTfBinary == "" {
+			payload.TerragruntTfBinary = OPENTOFU
+		}
+
+		if payload.TerragruntTfBinary == OPENTOFU && payload.OpentofuVersion == "" {
+			return errors.New("must supply opentofu version")
+		}
+
+		if payload.TerragruntTfBinary == TERRAFORM && payload.TerraformVersion == "" {
+			return errors.New("must supply terraform version")
+		}
+	} else {
+		payload.TerragruntTfBinary = ""
 	}
 
 	if payload.Type == OPENTOFU && payload.OpentofuVersion == "" {
 		return errors.New("must supply opentofu version")
 	}
 
-	if payload.TerragruntTfBinary != "" && payload.Type != TERRAGRUNT {
```

Comment:
not needed anymore.
We just ignore the value by doing:
`payload.TerragruntTfBinary = ""` (line 152 in the new code).

Reply:
are you sure? i think it is better to be explicit and tell the user to delete it

Reply:
sure I don't mind either way.

Reply:
so lets keep it, with test

Reply:
I actually already updated it. See latest commit.

Reply:
[6abd865](https://github.com/env0/terraform-provider-env0/pull/980/commits/6abd865082dad28d3a9787dd791a3a9e4747cbfb)